File: CPP/Program/Dosen.cpp
#include "Header.hh"
//DOSEN
Dosen::Dosen() : Civitas()
{

}

Dosen::Dosen(string nik, string name, char gender, string asal_universitas, string email_edu) : 
    Civitas(nik, name, gender, asal_universitas, email_edu)
{

}

Dosen::Dosen(string nik, string name, char gender, string asal_universitas, string email_edu, string nip, string fakultas, string pend_terakhir, string keahlian) :
    Civitas(nik, name, gender, asal_universitas, email_edu)
{
    this->nip = nip;
    this->fakultas = fakultas;
    this->pend_terakhir = pend_terakhir;
    this->keahlian = keahlian;
}

string Dosen::getNip()
{
    return nip;
}

void Dosen ::setNip(string nip)
{
    this->nip = nip;
}
 
string Dosen::getFakultas()
{
    return fakultas;
}

void Dosen ::setFakultas(string fakultas)
{
    this->fakultas = fakultas;
}

string Dosen::getPend_terakhir()
{
    return pend_terakhir;
}

void Dosen ::setPend_terakhir(string pend_terakhir)
{
    this->pend_terakhir = pend_terakhir;
}

string Dosen::getKeahlian()
{
    return keahlian;
}

void Dosen ::setKeahlian(string keahlian)
{
    this->keahlian = keahlian;
}

Dosen::~Dosen()
{

}

File: PYTHON/Program/Dosen.py
#DOSEN
from Civitas import Civitas
# from prodi import Prodi
class Dosen (Civitas):
    
    # Attributes dari class dosen
    __nip = ""
    __fakultas = ""
    __pend_terakhir = ""
    __keahlian = ""
    
    # Constructors
    # methods dari class dosen
    def __init__(self, nik = "", nama = "", jenis_kelamin ="", asal_universitas = "", email_edu = "", nip = "" , fakultas = "", pend_terakhir = "", keahlian = ""):
        super().__init__(nik, nama, jenis_kelamin, asal_universitas, email_edu)
        # Set attributes pada class dosen
        self.__fakultas = fakultas
        self.__nip = nip
        self.__pend_terakhir = pend_terakhir
        self.__keahlian = keahlian


    # Getter dan Setter
        
    # Getter dan setter untuk nim mahasiswa
    def get_nip(self):
        return self.__nip
    def set_nip(self, nip):
        self.__nip = nip
        
    # Getter dan setter untuk fakultas mahasiswa
    def get_fakultas(self):
        return self.__fakultas
    def set_fakultas(self, fakultas):
        self.__fakultas = fakultas
        
    # Getter dan setter untuk nik mahasiswa
    def get_pend_terakhir(self):
        return self.__pend_terakhir
    def set_pend_terakhir(self, pend_terakhir):
        self.__pend_terakhir = pend_terakhir
        
    # Getter dan setter untuk nama mahasiswa
    def get_keahlian(self):
        return self.__keahlian
    def set_keahlian(self, keahlian):
        self.__keahlian = keahlian

